### PR TITLE
Fix/renset

### PR DESCRIPTION
**Pull request by dzhwinter**
fix resnet, add fluid cifar dataset support

### CODE REVIEW

File: fluid/vgg16.py [hunk 1]
```diff
@@ -70,10 +70,16 @@ def conv_block(input, num_filter, groups, dropouts):
...
+            data_shape = [3, 32, 32]
+        else:
+            data_shape = [32, 32, 3]
     else:
         classdim = 102
-        data_shape = [3, 224, 224]
+        if args.data_format == 'NCHW':
+            data_shape = [3, 224, 224]
+        else:
+            data_shape = [224, 224, 3]
```
Comment (qingqing01) [inline fluid/vgg16.py]: The network is different for 224*224 input and 32*32 input, please refer to https://github.com/dzhwinter/benchmark/blob/master/tensorflow/resnet50.py#L360

Comment (dzhwinter) [reply fluid/vgg16.py]: done.

File: fluid/resnet50.py [hunk 2]
```diff
@@ -71,44 +77,44 @@ def print_arguments(args):
...
+    conv3 = conv_bn_layer(conv2, ch_out * 4, 1, 1, 0, act=None)
+    return fluid.layers.elementwise_add(x=short, y=conv3, act='relu')
+
+def layer_warp(block_func, input, ch_out, count, stride):
+    res_out = block_func(input, ch_out, stride)
+    for i in range(1, count):
+        res_out = block_func(res_out, ch_out, 1)
+    return res_out
+
+def resnet(input, class_dim, depth=50, data_format='NCHW'):
```
Comment (qingqing01) [inline fluid/resnet50.py]: 叫 `resnet_imagenet`，和TensorFlow里的配置对应吧 https://github.com/dzhwinter/benchmark/blob/master/tensorflow/resnet50.py#L220

File: fluid/resnet50.py [hunk 3]
```diff
@@ -71,44 +77,44 @@ def print_arguments(args):
         print('%s: %s' % (arg, value))
     print('------------------------------------------------')
 
+def conv_bn_layer(input, ch_out, filter_size, stride, padding, act='relu'):
+    tmp = fluid.layers.conv2d(
```
Comment (pkuyym) [inline fluid/resnet50.py]: Please avoid using `tmp`.

File: fluid/stacked_dynamic_lstm.py [hunk 4]
```diff
@@ -1,75 +1,94 @@
...
+                       stacked_num,
+                       class_num=2,
+                       is_train=True):
+    word_idx = fluid.layers.data(
+        name="word_idx", shape=[1], dtype="int64", lod_level=1)
+    embedding = fluid.layers.embedding(
+        input=word_idx, size=[dict_size, embedding_dim])
 
-    sentence = fluid.layers.fc(input=sentence, size=200, act='tanh')
+    sentence = fluid.layers.fc(input=embedding, size=hidden_dim * 4, bias_attr=True)
```
Comment (pkuyym) [inline fluid/stacked_dynamic_lstm.py]: Why multiply 4 here?

File: fluid/stacked_dynamic_lstm.py [hunk 5]
```diff
@@ -102,92 +121,98 @@ def gate_common(
...
         rnn.output(hidden)
 
-    last = fluid.layers.sequence_pool(rnn(), 'last')
-    logit = fluid.layers.fc(input=last, size=2, act='softmax')
-    loss = fluid.layers.cross_entropy(
-        input=logit,
-        label=fluid.layers.data(
-            name='label', shape=[1], dtype='int64'))
-    loss = fluid.layers.mean(x=loss)
+    lstm_out = fluid.layers.sequence_pool(input=rnn(), pool_type='max')
```
Comment (pkuyym) [inline fluid/stacked_dynamic_lstm.py]: Please double check 'max' or 'last' here.

File: fluid/resnet50.py (hunk 3 above)
Comment (dzhwinter) [reply fluid/resnet50.py]: done.

File: fluid/stacked_dynamic_lstm.py (hunk 5 above)
Comment (dzhwinter) [reply fluid/stacked_dynamic_lstm.py]: done

File: fluid/stacked_dynamic_lstm.py (hunk 4 above)
Comment (dzhwinter) [reply fluid/stacked_dynamic_lstm.py]: I thought there will be four cell inside DynamicRNN, so it should be multiplied with 4, fixed.

**Pull request merged**